TGControlPanel/ServerPage.cs

```diff
@@ -373,25 +373,68 @@ private void ServerGRestartButton_Click(object sender, EventArgs e)
 				return;
 			Server.GetComponent<ITGDreamDaemon>().RequestRestart();
 		}
+
+
 		private void FullUpdateWorker_DoWork(object sender, DoWorkEventArgs e)
 		{
-			var Updater = Server.GetComponent<ITGServerUpdater>();
+			var Repo = Server.GetComponent<ITGRepository>();
+			var DM = Server.GetComponent<ITGCompiler>();
 			switch (fuAction)
 			{
 				case FullUpdateAction.Testmerge:
-					updateError = Updater.UpdateServer(TGRepoUpdateMethod.None, false, (ushort)testmergePR);
+					updateError = Repo.MergePullRequest(testmergePR);
+					updateError = Repo.GenerateChangelog(out string error2);
+					if (updateError == null)
+						updateError = DM.Compile(true) ? null : "Compilation failed!";
 					break;
 				case FullUpdateAction.UpdateHard:
-					updateError = Updater.UpdateServer(TGRepoUpdateMethod.Hard, true);
+					updateError = Repo.Update(true);
+					updateError = Repo.GenerateChangelog(out error2);
```

Comment:
You shouldn't be stopping compiling if the change log fails to generate, its an optional thing, having python installed should be an optional thing.